pkg/apis/submariner/v1alpha1/submariner_types.go

```diff
@@ -60,16 +61,19 @@ type SubmarinerSpec struct {
 // SubmarinerStatus defines the observed state of Submariner
 // +k8s:openapi-gen=true
 type SubmarinerStatus struct {
-	NatEnabled                bool                    `json:"natEnabled"`
-	ColorCodes                string                  `json:"colorCodes,omitempty"`
-	ClusterID                 string                  `json:"clusterID"`
-	ServiceCIDR               string                  `json:"serviceCIDR,omitempty"`
-	ClusterCIDR               string                  `json:"clusterCIDR,omitempty"`
-	GlobalCIDR                string                  `json:"globalCIDR,omitempty"`
-	EngineDaemonSetStatus     *appsv1.DaemonSetStatus `json:"engineDaemonSetStatus,omitempty"`
-	RouteAgentDaemonSetStatus *appsv1.DaemonSetStatus `json:"routeAgentDaemonSetStatus,omitempty"`
-	GlobalnetDaemonSetStatus  *appsv1.DaemonSetStatus `json:"globalnetDaemonSetStatus,omitempty"`
-	Gateways                  *[]submv1.Gateway       `json:"gateways,omitempty"`
+	NatEnabled                  bool                      `json:"natEnabled"`
+	ColorCodes                  string                    `json:"colorCodes,omitempty"`
+	ClusterID                   string                    `json:"clusterID"`
+	ServiceCIDR                 string                    `json:"serviceCIDR,omitempty"`
+	ClusterCIDR                 string                    `json:"clusterCIDR,omitempty"`
+	GlobalCIDR                  string                    `json:"globalCIDR,omitempty"`
+	EngineDaemonSetStatus       *appsv1.DaemonSetStatus   `json:"engineDaemonSetStatus,omitempty"`
+	EngineContainerStatuses     *[]corev1.ContainerStatus `json:"engineContainerStatuses,omitempty"`
+	RouteAgentDaemonSetStatus   *appsv1.DaemonSetStatus   `json:"routeAgentDaemonSetStatus,omitempty"`
+	RouteAgentContainerStatuses *[]corev1.ContainerStatus `json:"routeAgentContainerStatuses,omitempty"`
```

Comment:
I see that in https://github.com/submariner-io/submariner/issues/777 we are talking about the status of the pods. The Daemonset status already provides numbers on Desired, Current, Ready, Up-to-date, Available. Isn't that enough?

Reply:
This is an initial stab at getting the information; I’ll be aggregating it later but I want to get a feel for what we see in various situations. In relation to submariner-io/submariner#777, the goal is to provide more detailed information on `DaemonSet` failures, *e.g.* containers in `CrashLoopBackoff` state. The information here is also potentially useful for `subctl show versions` etc.

Reply:
I think it's ok to track the pods, but not to post it (un-aggregated) into the Status of submariner. That's going to be a huge scalability performance. It means that for any Pod status change we will be triggering an update to the Submariner status in API. 

And it will be a big blob for big deployments, Imagine a 1000 node deployment... that's one pod per node. I inspected the json size of containerstatus field in json and it was ranging from 1k to 4k. would mean >1MB  <4MB API request every time we try to update the Submariner object (with gateway updates or anything else...). 

Also the output yaml would become huge.

Is it ok if we start watching those objects (which IMO makes sense), but not publishing raw status of all of them?, we can follow up with more patches to aggregate status?

I know this probably sounds like early optimization, but I have been bitten by stuff like this in the past (openstack/neutron doing periodic reports with huge blobs eventually killing the system).